app.py

```diff
@@ -68,12 +93,149 @@
     _LOGGER.info("installation=%s", RUNTIME_CONTEXT.app_installation)
 
 
+@process_event_actions("pull_request", {"opened", "reopened", "synchronize", "edited"})
+@process_webhook_payload
+async def on_pr_open_or_sync(*, action, number, pull_request, repository, sender, installation, **kwargs):
+    """React to an opened or changed PR event.
+
+    Send a status update to GitHub via Checks API.
+    """
+    _LOGGER.info(f"on_pr_open_or_sync: working on PR {pull_request['html_url']}",)
+
+    github_api = RUNTIME_CONTEXT.app_installation_client
+
+    pr_head_sha = pull_request["head"]["sha"]
+    base_repo_url = pull_request["base"]["repo"]["url"]
+    repo_url = pull_request["head"]["repo"]["html_url"]
+    check_runs_base_uri = f"{base_repo_url}/check-runs"
+
+    if pr_head_sha is None:
+        _LOGGER.error(f"on_pr_open_or_sync: no Pull Request head sha found, stopped working!")
+        return
+
+    _LOGGER.info(f"on_pr_open_or_sync: repo_url {repo_url} will be used for check-run")
+    _LOGGER.info(f"on_pr_open_or_sync: PR commit id {pr_head_sha} will be used for check-run")
+
+    resp = await github_api.post(
+        check_runs_base_uri,
+        preview_api_version="antiope",
+        data={
+            "name": CHECK_RUN_NAME,
+            "head_sha": pr_head_sha,
+            "status": "queued",
+            "started_at": f"{datetime.utcnow().isoformat()}Z",
+        },
+    )
+
+    check_run_id = int(resp["id"])  # TODO do we need some marshaling here?
+
+    check_runs_updates_uri = f"{check_runs_base_uri}/{check_run_id}"
+    _LOGGER.info(f"on_pr_open_or_sync: check_run_id: {check_run_id}")
+
+    data = {
+        "github_event_type": "thoth_thamos_advise",
+        "github_check_run_id": check_run_id,
+        "github_installation_id": installation["id"],
+        "origin": repo_url,
+        "revision": pr_head_sha,
+    }
+    async with aiohttp.ClientSession() as session:
+        resp = await session.post(USER_API_URL, json=data)
+        _LOGGER.info(f"on_pr_open_or_sync: user-api resp: {resp}")
+
+    resp = await github_api.patch(
+        check_runs_updates_uri, preview_api_version="antiope", data={"name": CHECK_RUN_NAME, "status": "in_progress"},
```

Comment:
TODO we should have a timeout for this, just to keep the github check status sane
@pacospace